### PR TITLE
Fix MIDIIn.connectAll on Linux to only connect to external sources and not internal ones

### PR DESCRIPTION
**Pull request by sensestage**
Fix for issue #1487.

### CODE REVIEW

File: SCClassLibrary/Common/Control/MIDIOut.sc [hunk 1]
```diff
@@ -248,12 +257,21 @@ MIDIIn {
...
 		^MIDIClient.sources.detect({ |endPoint| endPoint.device == deviceName and: {endPoint.name == portName}});
 	}
+
+	// *connectAll {
+	// 	if(MIDIClient.initialized.not,{ MIDIClient.init });
+	// 	MIDIClient.sources.do({ |src,i|
+	// 		MIDIIn.connect(i,src);
+	// 	});
+	// }
+
```
Comment (scztt) [inline SCClassLibrary/Common/Control/MIDIOut.sc]: Delete instead of commenting out?

File: lang/LangPrimSource/SC_AlsaMIDI.cpp [hunk 2]
```diff
@@ -1008,6 +1012,18 @@ int prSendSysex(VMGlobals *g, int numArgsPushed)
...
+  
+  PyrSlot* args = g->sp;
+  if (!gMIDIClient.mHandle) return errFailed;
+
+  
+  SetInt(args, gMIDIClient.mClientID );
+  
+  return errNone;
+}
+
```
Comment (scztt) [inline lang/LangPrimSource/SC_AlsaMIDI.cpp]: This only handles the linux case, no? There needs to be an OSX and Win implementation of this primitive. The OSX would look something like this (I think...following SC_CoreMIDI.cpp):

		SInt32 id;
		MIDIObjectGetIntegerProperty(gMIDIClient, kMIDIPropertyUniqueID, &id);

**Comment by sensestage** [comment]
It only handles the Linux case, and it is also only needed for the Linux case.
Linux has a patchbay approach to MIDI with Alsa, meaning that each application creates input and outputs and you can patch between these. On Windows and OSX this is not possible. So a fix is not needed for these platforms. In my fix, I also only call the primitive from Linux, not from OSX or Windows.

File: SCClassLibrary/Common/Control/MIDIOut.sc (hunk 1 above)
Comment (sensestage) [inline SCClassLibrary/Common/Control/MIDIOut.sc]: yes, agreed.

**Pull request merged**